libbeat/cmd/instance/beat.go

```diff
@@ -435,7 +448,7 @@ func (b *Beat) TestConfig(bt beat.Creator) error {
 }
 
 // Setup registers ES index template, kibana dashboards, ml jobs and pipelines.
-func (b *Beat) Setup(bt beat.Creator, template, setupDashboards, machineLearning, pipelines, policy bool) error {
+func (b *Beat) Setup(bt beat.Creator, flags map[string]bool) error {
```

Comment:
I have mixed feeling about the strategy used to define a variable amount of options, using a map[string]bool limit ourself to boolean flags. 

I would prefer one of the following approach:

- Using a Settings Struct, It was the advantage of allowing more complex data types and it keep type safety when we are defining the values.

```golang
type Settings struct {
    //TemplateKey used for defining template in setup cmd
	TemplateKey bool
	//DashboardKey used for registering dashboards in setup cmd
	DashboardKey bool
	//MachineLearningKey used for registering ml jobs in setup cmd
	MachineLearningKey bool
	//PipelineKey used for registering pipelines in setup cmd
	PipelineKey bool
	//ILMPolicyKey used for registering ilm in setup cmd
	ILMPolicyKey bool
}
```

- Or use a more functional way with a combination of a settings struct an `Option` type used with a variadic definition.

```golang
type Settings struct {
    //TemplateKey used for defining template in setup cmd
	TemplateKey bool
	//DashboardKey used for registering dashboards in setup cmd
	DashboardKey bool
	//MachineLearningKey used for registering ml jobs in setup cmd
	MachineLearningKey bool
	//PipelineKey used for registering pipelines in setup cmd
	PipelineKey bool
	//ILMPolicyKey used for registering ilm in setup cmd
	ILMPolicyKey bool
}

type Option func(s *settings) {}

func WithTemplate(s *settings) {
 s.TemplateKey = true
}

func WithDashboard(s *settings) {
 s.Dashboard= true
}


func (b *Beat) Setup(bt beat.Creator, options ...Option) error {
 s := settings{}
 for _, opt := range options {
   opt(s)
 }

 if s.Dashboard {
   //...
 }
}
```

The advantages of using a Struct we could possibly add predicates on the setting type.